### ISSUE TITLE
Prisma generate fails when enum has duplicates that are mapped

### ISSUE THREAD

**Issue by divyenduz** [prisma/prisma#3010]
## Bug description

Such a schema can be produced by introspection. 

## How to reproduce

1. Run the following SQL in a MySQL database (Other databases untested)

```sql
CREATE TABLE `e` (
  `id` bigint(20) unsigned NOT NULL,
  `visible` enum('301-product','302-product') NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=MyISAM DEFAULT CHARSET=utf8;
```

2. Introspect this database

```prisma
model e {
  id      Int       @id
  visible e_visible
}

enum e_visible {
  product @map("301-product")
  product @map("302-product")
}
```

3. Run `prisma generate` and observe the error

```
divyendusingh [p2-enum-map-dupe]$ yarn prisma generate
yarn run v1.22.4
$ /Users/divyendusingh/Documents/prisma/p2-enum-map-dupe/node_modules/.bin/prisma generate
Error: Schema parsing
error: Value "product" is already defined on enum "e_visible".
  -->  schema.prisma:13
   |
12 |   product @map("301-product")
13 |   product @map("302-product")
14 | }
   |

Validation Error Count: 1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
divyendusingh [p2-enum-map-dupe]$ unset DEBUG                                                                                                 1 ↵
divyendusingh [p2-enum-map-dupe]$ yarn prisma generate
yarn run v1.22.4
$ /Users/divyendusingh/Documents/prisma/p2-enum-map-dupe/node_modules/.bin/prisma generate
Error: Schema parsing
error: Value "product" is already defined on enum "e_visible".
  -->  schema.prisma:13
   |
12 |   product @map("301-product")
13 |   product @map("302-product")
14 | }
   |

Validation Error Count: 1
error Command failed with exit code 1.

```

This can be fixed by manually adjusting the schema to resolve collisions

```
enum e_visible {
  product_1 @map("301-product")
  product_2 @map("302-product")
}
```

## Expected behavior

Since we are mapping, we can resolve this by not having collisions in the enum values. 

## Prisma information

```
yarn run v1.22.4
$ /Users/divyendusingh/Documents/prisma/p2-enum-map-dupe/node_modules/.bin/prisma --version
@prisma/cli          : 2.3.0-dev.16
Current platform     : darwin
Query Engine         : query-engine 21bd701db858e37b8f3157190a86da93e21ec5df (at /Users/divyendusingh/Documents/prisma/p2-enum-map-dupe/node_modules/@prisma/cli/query-engine-darwin)
Migration Engine     : migration-engine-cli 21bd701db858e37b8f3157190a86da93e21ec5df (at /Users/divyendusingh/Documents/prisma/p2-enum-map-dupe/node_modules/@prisma/cli/migration-engine-darwin)
Introspection Engine : introspection-core 21bd701db858e37b8f3157190a86da93e21ec5df (at /Users/divyendusingh/Documents/prisma/p2-enum-map-dupe/node_modules/@prisma/cli/introspection-engine-darwin)
Format Binary        : prisma-fmt 21bd701db858e37b8f3157190a86da93e21ec5df (at /Users/divyendusingh/Documents/prisma/p2-enum-map-dupe/node_modules/@prisma/cli/prisma-fmt-darwin)
Done in 1.66s.

```

Test utils databases affected: 

- mysql/prestashop

**Comment by janpio** [comment]
We currently do not want to add opinionation on this edge case of duplications caused by our (fully unopinionated) renaming logic. So while this is something that is not optimal right now, changing this is not a priority.

**Comment by divyenduz** [comment]
Note: Prisma VSCode extension catches this nicely

![image](https://user-images.githubusercontent.com/746482/87312723-9106af00-c53e-11ea-9006-570f659a91c1.png)

**Comment by tomhoule** [comment]
Pending on a principled stance on identifiers in PSL.

**Comment by pantharshit00** [comment]
We are still waiting here for https://github.com/prisma/prisma/issues/6680

**Comment by janpio** [comment]
What is the connection to named constraints here @pantharshit00?

Note from workshop: We should check if this already creates an Introspection warning, if not this would be a good use case.

**Comment by pantharshit00** [comment]
@janpio Looks like I referenced the wrong thing when Tom said "identifiers" in his message.

**Comment by tomhoule** [comment]
Do we still consider this a bug? I don't see us changing this.

**Comment by janpio** [comment]
Nope. This is definitely intended behavior in our current understanding.
